File: ts/project/.sdk/tm/php/utility/PrepareParams.php
<?php
declare(strict_types=1);

// ProjectName SDK utility: prepare_params

class ProjectNamePrepareParams
{
    public static function call(ProjectNameContext $ctx): array
    {
        $utility = $ctx->utility;
        $point = $ctx->point;
        $params = [];
        if ($point) {
            $args = \Voxgig\Struct\Struct::getprop($point, 'args');
            if (is_array($args)) {
                $p = \Voxgig\Struct\Struct::getprop($args, 'params');
                if (is_array($p)) {
                    $params = $p;
                }
            }
        }
        $out = [];
        foreach ($params as $pd) {
            $val = ($utility->param)($ctx, $pd);
            if ($val !== null && is_array($pd)) {
                $name = \Voxgig\Struct\Struct::getprop($pd, 'name');
                if (is_string($name) && $name !== '') {
                    $out[$name] = $val;
                }
            }
        }
        return $out;
    }
}

File: ts/project/.sdk/tm/rb/utility/prepare_params.rb
# ProjectName SDK utility: prepare_params
require_relative 'struct/voxgig_struct'
module ProjectNameUtilities
  PrepareParams = ->(ctx) {
    utility = ctx.utility
    point = ctx.point
    params = []
    if point
      args = VoxgigStruct.getprop(point, "args")
      if args.is_a?(Hash)
        p = VoxgigStruct.getprop(args, "params")
        params = p if p.is_a?(Array)
      end
    end
    out = {}
    params.each do |pd|
      val = utility.param.call(ctx, pd)
      if val && pd.is_a?(Hash)
        name = VoxgigStruct.getprop(pd, "name")
        out[name] = val if name.is_a?(String) && !name.empty?
      end
    end
    out
  }
end
